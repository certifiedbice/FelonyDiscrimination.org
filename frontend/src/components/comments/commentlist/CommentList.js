import React,{Component} from 'react'
import Comment from '../comment/Comment';
import './CommentList.css';

export default class CommentList extends Component{
	render(){
		return(
			<div id="comment-list-container">
				<div>{/*comments: total information here*/}</div>
				{this.props.comments.map((comment,index)=><Comment key={index} comment={comment}/>)}
            </div>
		);
	}
}


                