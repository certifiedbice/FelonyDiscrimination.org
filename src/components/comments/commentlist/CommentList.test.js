import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CommentList from './CommentList';

const comments=[
    {},
    {},
];

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<CommentList comments={comments}/>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<CommentList comments={comments}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});