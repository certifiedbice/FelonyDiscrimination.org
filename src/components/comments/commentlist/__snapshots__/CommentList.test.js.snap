// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="comment-list-container"
>
  <div />
  <div
    className="comment-item"
  >
    <div
      id="comment-date_published"
    >
      undefined undefined NaN NaN
    </div>
    <div
      id="comment-title"
    >
      <h3>
        User says: 
      </h3>
    </div>
    <div
      id="comment-content"
    />
  </div>
  <div
    className="comment-item"
  >
    <div
      id="comment-date_published"
    >
      undefined undefined NaN NaN
    </div>
    <div
      id="comment-title"
    >
      <h3>
        User says: 
      </h3>
    </div>
    <div
      id="comment-content"
    />
  </div>
</div>
`;
